### ISSUE TITLE
TypeError: type.getFields is not a function

### ISSUE THREAD

**Issue by arondesign** [avantcredit/gql2ts#65]
Hi there, great work on this project so far! I ran across an issue in typeToInterface in fromSchema/index.ts. If a GraphQLUnionType is used in the schema, it throws `TypeError: type.getFields is not a function` at Line 173 in :

```
    let isInput: boolean = type instanceof GraphQLInputObjectType;
    const f1: GraphQLInputFieldMap | GraphQLFieldMap<any, any> =
      (type instanceof GraphQLInputObjectType) ? type.getFields() : ((type as GraphQLObjectType).getFields)();
    let f: Array<GraphQLField<any, any> | GraphQLInputField> = Object.keys(f1).map(k => f1[k]);
```

**Comment by brettjurgens** [comment]
nice, thanks for the report.

do you have a minimal reproduction so I can take a further look?

**Comment by arondesign** [comment]
Sure, try running the schema.json gist I posted up at: https://gist.github.com/arondesign/acab6ce16b8a2128e6431de6c6a9f0b4

**Comment by brettjurgens** [comment]
thanks, found the issue 😄 #66

**Comment by brettjurgens** [comment]
@arondesign should be fixed in v1.0.1

**Comment by arondesign** [comment]
Awesome!

**Comment by arondesign** [comment]
Hi @brettjurgens, I think this still needs one adjustment. If you run the schema.json file I sent you, you'll notice in

```
  interface IQuery {
    __typename: string;
    Products: IProducts | null;
    ProductRoute: IProductUnion | null;
    Product: IProduct | null;
  }
```

IProducts is undefined. However, the following is generated

```
  type ConfigurationInterface = IHaloEarrings | IHaloRing;

  interface IConfigurationInterface {
    __typename: string;
    attributes: any | null;
    components: any | null;
    id: string | null;
    kind: string | null;
    images: Array<string> | null;
    price: number | null;
    msrp: number | null;
    hero: string | null;
    heroSecondary: string | null;
    name: string | null;
    thumbnail: string | null;
    description: string | null;
    route: string | null;
    shortDescription: string | null;
    configuration: any | null;
    parts: any | null;
    video: string | null;
    relatedOptions: IRelatedOptions | null;
  }
```

The unioned types do share a common interface (ConfigurationInterface), but the output for
```
  type ConfigurationInterface = IHaloEarrings | IHaloRing;
```

should instead read
```
  type IProductUnion = IHaloEarrings | IHaloRing;
```

Or of course, just "ProductUnion" for both.

**Comment by brettjurgens** [comment]
whoops. don't know what i was thinking there. fixed in #69

**Comment by brettjurgens** [comment]
@arondesign 1.0.2 should work, i ran it against your schema and it looked good
